### PR TITLE
histogram anomaly detection

### PR DESCRIPTION
**Pull request by rupnikj**
- refactored modinit and binding.gyp (big object compilation problems on windows)
- tracing number of wrapped objects (nodejs code: `qm.stats()`)
- histogram andomaly detection
- kernel density estimation
- autmatic bandwidth selection
- geo-spatial module (stay point detector API, not implemented yet)

### CODE REVIEW

File: src/glib/base/xmath.h [hunk 1]
```diff
@@ -493,7 +495,7 @@ class THist {
 	double GetBucketMn(const int& BucketN) const { return MnVal + BucketN * BucketSize; }
```
Comment (blazf) [inline src/glib/base/xmath.h]: Tabs -> spaces

File: src/nodejs/analytics/analytics.h [hunk 2]
```diff
@@ -8,6 +8,14 @@
 #ifndef ANALYTICS_H_
 #define ANALYTICS_H_
 
+#include <node.h>
+#include <node_object_wrap.h>
+#include "../nodeutil.h"
+#include "fs_nodejs.h"
+#include "la_nodejs.h"
+#include "qminer_ftr.h"
```
Comment (blazf) [inline src/nodejs/analytics/analytics.h]: Why does analytics.h depand on qminer?

File: src/qminer/qminer_aggr.cpp [hunk 3]
```diff
@@ -2012,5 +2012,389 @@ TRecFilterAggr::TRecFilterAggr(const TWPt<TBase>& Base, const PJsonVal& ParamVal
...
+        Thresholds.Add(0.05);
+        Thresholds.Add(0.01);
+        Thresholds.Add(0.001);
+    }
+    // Other models that could be implemented:
+    //   laplace smoothing
+    //   fit 1 gaussian
+}
+
+void THistogramToPMFModel::ClassifyAnomalies(const TFltV& PMF, TFltV& Severities) {
```
Comment (blazf) [inline src/qminer/qminer_aggr.cpp]: Would it make sense to move this also to glib, e.g. `TKDEModel`?

File: src/qminer/qminer_aggr.cpp [hunk 4]
```diff
@@ -2012,5 +2012,389 @@ TRecFilterAggr::TRecFilterAggr(const TWPt<TBase>& Base, const PJsonVal& ParamVal
...
+	Obj->AddToObj("explain", Explanation);
+    Obj->AddToObj("thresholds", TJsonVal::NewArr(Model.Thresholds));
+    Obj->AddToObj("tol", Model.Tol);
+    return Obj;
+}
+
+//////////////////////////////////////////////////////////////////////////////////
+/// Kernel-Density-Estimation
+
+double TKDEModel::EvalKernel(const double& Data, const double& Eval, const double& Bandwidth, const THistogramPMFModelType& KType) {
```
Comment (blazf) [inline src/qminer/qminer_aggr.cpp]: I would move actual implementation out of qminer into glib/mine/

File: src/qminer/qminer_aggr.h [hunk 5]
```diff
@@ -1410,17 +1408,22 @@ class TOnlineHistogram : public TStreamAggr, public TStreamAggrOut::IFltVec {
     /// serilization to JSon
     PJsonVal SaveJson(const int& Limit) const { return Model.SaveJson(); }
 
-    /// returns the number of bins 
+    /// Finds the bin index given val, returns -1 if not found
+    int FindBin(const double& Val) const { return Model.FindBin(Val); }
+	/// Returns the bound
+	double GetBoundN(const int& BoundN) const { return Model.GetBoundN(BoundN); }
```
Comment (blazf) [inline src/qminer/qminer_aggr.h]: Tabs -> spaces

File: src/qminer/qminer_aggr.h [hunk 6]
```diff
@@ -1771,11 +1774,174 @@ class TRecFilterAggr : public TStreamAggr {
     TStr Type() const { return GetType(); }
 };
 
+typedef enum {
```
Comment (blazf) [inline src/qminer/qminer_aggr.h]: This should all move out of qminer to glib/mine, preferably to its own namespace?

File: src/nodejs/analytics/analytics.h (hunk 2 above)
Comment (rupnikj) [inline src/nodejs/analytics/analytics.h]: I changed it to qminer_core. SVM uses stuff like TQm::TEnv::Error for logging

File: src/glib/base/xmath.h (hunk 1 above)
Comment (rupnikj) [inline src/glib/base/xmath.h]: fixed

File: src/qminer/qminer_aggr.cpp (hunk 3 above)
Comment (rupnikj) [inline src/qminer/qminer_aggr.cpp]: moved

File: src/qminer/qminer_aggr.cpp (hunk 4 above)
Comment (rupnikj) [inline src/qminer/qminer_aggr.cpp]: moved

File: src/qminer/qminer_aggr.h (hunk 5 above)
Comment (rupnikj) [inline src/qminer/qminer_aggr.h]: fixed

File: src/qminer/qminer_aggr.h (hunk 6 above)
Comment (rupnikj) [inline src/qminer/qminer_aggr.h]: moved

**Pull request merged**